packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart

```diff
@@ -75,14 +73,17 @@ class VMDeveloperToolsScreenBody extends StatefulWidget {
 
 class _VMDeveloperToolsScreenState extends State<VMDeveloperToolsScreenBody>
     with AutoDisposeMixin {
-  VMDeveloperToolsController controller;
+  bool initialized = false;
+  late VMDeveloperToolsController controller;
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
     final newController = Provider.of<VMDeveloperToolsController>(context);
-    if (newController == controller) return;
+    // Don't access `controller` if we haven't already initialized it.
+    if (initialized && newController == controller) return;
     controller = newController;
+    initialized = true;
```

Comment:
FYI @polina-c we should use this pattern elsewhere when we are initializing `controller` in widget state classes. (https://github.com/flutter/devtools/pull/3822#discussion_r823183816)

Reply:
So, the idea is to set it just once, right?
Good to know thank you. Thank you!

Reply:
We will still update `controller` if the dependencies changed (i.e. if the provided controller changed), but with this pattern we will not access `controller` to check equality before it is initialized, thus allowing it to be late and non-nullable.